src/GafferScene/SetAlgo.cpp

```diff
@@ -218,13 +220,41 @@ struct AstEvaluator
 	typedef PathMatcher result_type;
 
 	AstEvaluator( const ScenePlug *scene )
+		: m_scene( scene ), m_wildcardExpression( "\\*" )
 	{
-		m_scene = scene;
 	}
 
 	result_type operator()( const SetName &set ) const
 	{
-		return m_scene->set( set.name )->readable();
+		if( !boost::regex_search( set.name, m_wildcardExpression ) )
+		{
+			return m_scene->set( set.name )->readable();
+		}
+
+		result_type result;
+
+		const std::vector<IECore::InternedString> &setNames = m_scene->setNamesPlug()->getValue()->readable();
```

Comment:
This will crash one day - you need to be holding a reference to the owning object via a `Ptr`.